framework/wazuh/decoder.py

```diff
@@ -354,7 +354,13 @@ def upload_decoder_file(filename: str, content: str, relative_dirname: str = Non
             raise WazuhError(1905)
         elif overwrite and exists(full_path):
             backup_file = f'{full_path}.backup'
-            delete_file_with_backup(backup_file, full_path, delete_decoder_file)
+            try:
+                full_copy(full_path, backup_file)
+            except IOError as exc:
+                raise WazuhError(1019) from exc
+
+            delete_decoder_file(filename=filename,
+                                relative_dirname=relative_dirname)
```

Comment:
I just realized our RBAC reference documentation is wrong. `PUT /decoders/files/{filename}` requires `decoders:update` and `decoders:delete` RBAC actions, but only `decoders:update` is listed:
- https://documentation.wazuh.com/current/user-manual/api/rbac/reference.html#decoders-update

The same happens for rules too. Could you open an issue in wazuh-documentation to fix this in the future?